tests/user_examples/extra_src/shared/wetting_coupled_spatial_temporal_method.h

```diff
@@ -0,0 +1,104 @@
+/* ------------------------------------------------------------------------- *
+ *                                SPHinXsys                                  *
+ * ------------------------------------------------------------------------- *
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle *
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for    *
+ * physical accurate simulation and aims to model coupled industrial dynamic *
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH   *
+ * (smoothed particle hydrodynamics), a meshless computational method using  *
+ * particle discretization.                                                  *
+ *                                                                           *
+ * SPHinXsys is partially funded by German Research Foundation               *
+ * (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1,            *
+ *  HU1527/12-1 and HU1527/12-4.                                             *
+ *                                                                           *
+ * Portions copyright (c) 2017-2023 Technical University of Munich and       *
+ * the authors' affiliations.                                                *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain a *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+ *                                                                           *
+ * ------------------------------------------------------------------------- */
+/**
+ * @file 	fluid_surface_complex.h
+ * @brief 	Here, we define the algorithm classes for fluid dynamics within the body.
+ * @details We consider here weakly compressible fluids. The algorithms may be
+ * 			different for free surface flow and the one without free surface.
+ * @author	Chi Zhang and Xiangyu Hu
+ */
+
+#ifndef WETTING_COUPLED_SPATIAL_TEMPORAL_COMPLEX_H
+#define WETTING_COUPLED_SPATIAL_TEMPORAL_COMPLEX_H
+
+#include "fluid_surface_complex.h"
+
+namespace SPH
+{
+namespace fluid_dynamics
+{
+/**
+ * @class WettingCoupledFreeSurfaceIndicationComplex
+ * @brief indicate the interfacial fluid particles near the fluid-structure intersurface.
+ */
+class WettingCoupledFreeSurfaceIndicationComplex : public FreeSurfaceIndicationComplex
```

Comment:
FreeSurfaceWettingIndicationComplex maybe a better term.

Reply:
ok, thanks for your good suggestion, I will discuss this suggested term name with Prof,HU, then make it sure about the new class name

Reply:
Dr. CHI,  with Fabien' and your suggestion, I have disscused with Prof.hu again.  We give a new name to the class and a discription to it, as blow. Please check
@Class NonWettingSurfaceIndication
@brief Non wetting surface particles include free-surface ones and interfacial ones near the non-wetted structure.
@brief Even the position divergence of interfacial fluid pariticles has satisfied with the threshold of spatial-temporal
identification approach to be identified as internal ones,they will remain as free-surface ones if without
any wetted neighboring solid particles.
*/
class NonWettingSurfaceIndication : public FreeSurfaceIndicationComplex